Foundation/NSData.swift

```diff
@@ -431,9 +431,22 @@ extension NSData {
         }
         return false
     }
-    
+
+    ///    Write the contents of the receiver to a location specified by the given file URL.
+    ///
+    ///    - parameter url:              The location to which the receiver’s contents will be written.
+    ///    - parameter writeOptionsMask: An option set specifying file writing options.
+    ///
+    ///    - throws: This method returns Void and is marked with the `throws` keyword to indicate that it throws an error in the event of failure.
+    ///
+    ///      This method is invoked in a `try` expression and the caller is responsible for handling any errors in the `catch` clauses of a `do` statement, as described in [Error Handling](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html#//apple_ref/doc/uid/TP40014097-CH42) in [The Swift Programming Language](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/index.html#//apple_ref/doc/uid/TP40014097) and [Error Handling](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/BuildingCocoaApps/AdoptingCocoaDesignPatterns.html#//apple_ref/doc/uid/TP40014216-CH7-ID10) in [Using Swift with Cocoa and Objective-C](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/BuildingCocoaApps/index.html#//apple_ref/doc/uid/TP40014216).
     public func writeToURL(url: NSURL, options writeOptionsMask: NSDataWritingOptions) throws {
-        NSUnimplemented()
+        guard let path = url.path where url.filePathURL == true else {
```

Comment:
Building on linux i get this failure
```
Foundation/NSData.swift:444:45: error: cannot convert value of type 'NSURL?' to expected argument type 'Bool'
        guard let path = url.path where url.filePathURL == true else {
                                        ~~~~^~~~~~~~~~~
```
which is really strange since i don't get that on mac